Middleware/Services/BLE_Service/BLE_Service.c

```diff
@@ -0,0 +1,714 @@
+        }
+        break;
+
+        case BLE_ADM_CMD_RX:
+        {
+            /* Received user input on User Command characteristic. Notify Registration application.
+               Note: Data must be preserved until the Registration application receives and
+               processes it. */
+            Ble_tstrRxData *pstrRxData = (Ble_tstrRxData *)malloc(sizeof(Ble_tstrRxData));
+
+            if (pstrRxData)
+            {
+                pstrRxData->pu8Data = (uint8_t *)malloc(pstrEvent->strRxData.u16Length + 1);
+                pstrRxData->u16Length = pstrEvent->strRxData.u16Length;
+
+                /* Successfully allocated memory for data pointer */
+                if (NULL == pstrRxData->pu8Data)
+                {
+                    /* Free allocated memory */
+                    free(pstrRxData);
+                }
+
+                /* Copy data into buffer and dispatch it to the Registration application */
+                memcpy((void *)pstrRxData->pu8Data, pstrEvent->strRxData.pu8Data, pstrRxData->u16Length);
+                (void)AppMgr_enuDispatchEvent(BLE_ADM_USER_INPUT_RECEIVED, (void *)pstrRxData);
+            }
+        }
+        break;
+
+        default:
+            /* Nothing to do */
+            break;
+        }
+    }
+}
+static void vidCtsEventHandler(ble_cts_c_t *pstrCtsInstance, ble_cts_c_evt_t *pstrEvent)
+{
+    /* Make sure valid arguments are passed */
+    if(pstrCtsInstance && pstrEvent)
+    {
+        switch(pstrEvent->evt_type)
+        {
+        case BLE_CTS_C_EVT_DISCOVERY_COMPLETE:
+        {
+            /* Current Time Service discovered on Central's GATT server and link has been
+               established. Associate established link to this instance of the CTS client by
+               assigning connection and characteristic handles to it */
+            (void)ble_cts_c_handles_assign(&BleCtsInstance,
+                                           pstrEvent->conn_handle,
+                                           &pstrEvent->params.char_handles);
+            /* Set Current Time reading flag */
+            bTimeReadingPossible = true;
+        }
+        break;
+
+        case BLE_CTS_C_EVT_DISCOVERY_FAILED:
+        {
+            /* Clear Current Time reading flag */
+            bTimeReadingPossible = false;
+        }
+        break;
+
+        case BLE_CTS_C_EVT_CURRENT_TIME:
+        {
+            /* Invoke Attribution application's current time data callback */
+            pfCtsCallback(&pstrEvent->params.current_time.exact_time_256);
+        }
+        break;
+
+        default:
+            /* Nothing to do */
+            break;
+        }
+    }
+}
+
+static void vidCtsErrorHandler(uint32_t u32Error)
+{
+    APP_ERROR_HANDLER(u32Error);
+}
+
+static void vidDataBaseDiscHandler(ble_db_discovery_evt_t *pstrEvent)
+{
+    /* Invoke database discovery handler to handle Current Time Service related events */
+    ble_cts_c_on_db_disc_evt(&BleCtsInstance, pstrEvent);
+}
```

Comment:
Leave a line. There's a handy little tool called "uncrustify" where you can define a general code layout file containing stylistic rules you'd like your project to abide by and it will enforce them for you with just the call of a command. You can install it on all platform as well but it's usually far easier to use on Linux. This will save you the headache of having to stay on the lookout for little aesthetic inconsistencies like this.